imap/jmap_mail.c

```diff
@@ -11153,7 +11169,14 @@ static int _email_bulkupdate_plan_mailboxids(struct email_bulkupdate *bulk, ptra
                 else {
                     /* Delete message from mailbox */
                     ptrarray_append(&plan->delete, uidrec);
-                    plan->needrights |= JACL_REMOVEITEMS;
+
+                    /* Don't require rights to remove items from the
+                       normally read-only $scheduled mailbox
+                       since we have already verified that the emails
+                       in this plan are indeed scheduled for sending */
+                    if (strcmpnull(scheduled_uniqueid, mboxrec->mbox_id)) {
+                        plan->needrights |= JACL_REMOVEITEMS;
```

Comment:
The comment would warrant this to be written as

    if (!strcmpnull(scheduled_uniqueid, mbox_id))
        plan->needrights &= ~JACL_REMOVEITEMS;

Reply:
I understand your comment, but we haven't added JACL_REMOVEITEMS at this point and we DO want it for non-scheduled mailboxes.  I will try to fix the comment

Reply:
Clarified the logic and removing confusing comments